components/forecast/WeatherTab.tsx

```diff
@@ -0,0 +1,114 @@
+import {Card} from 'components/content/Card';
+import {Center, HStack, View, VStack} from 'components/core';
+import {useNWACWeatherForecast} from 'components/forecast/useNWACWeatherForecast';
+import {AllCapsSm, AllCapsSmBlack, Body, BodyBlack, BodySemibold, Title3Black} from 'components/text';
+import {ActivityIndicator, StyleSheet} from 'react-native';
+import {colorLookup} from 'theme';
+import {AvalancheForecastZone} from 'types/nationalAvalancheCenter';
+
+interface WeatherTabProps {
+  zone: AvalancheForecastZone;
+}
+
+const timeOfDayString = (period: 'day' | 'night', subperiod: 'early' | 'late') => {
+  if (period === 'day') {
+    return subperiod === 'early' ? 'morning' : 'afternoon';
+  } else {
+    return subperiod === 'early' ? 'evening' : 'overnight';
+  }
+};
+
+export const WeatherTab: React.FC<WeatherTabProps> = ({zone}) => {
+  const {isLoading, isError, data: forecast} = useNWACWeatherForecast();
+
+  if (isLoading) {
+    return (
+      <Center style={StyleSheet.absoluteFillObject}>
+        <ActivityIndicator />
+      </Center>
+    );
+  } else if (isError) {
+    return (
+      <Center style={StyleSheet.absoluteFillObject}>
+        <Body>Could not fetch weather forecast for zone {zone.name}</Body>
+      </Center>
+    );
+  }
+
+  return (
+    <VStack space={8} bgColor={'#f0f2f5'}>
+      <Card marginTop={1} borderRadius={0} borderColor="white" header={<Title3Black>Weather Forecast</Title3Black>}>
+        <HStack justifyContent="space-evenly" alignItems="flex-start" space={8}>
+          <VStack space={8} style={{flex: 1}}>
+            <AllCapsSmBlack>Issued</AllCapsSmBlack>
+            <AllCapsSm style={{textTransform: 'none'}} color="lightText">
+              {forecast.published_time}
+            </AllCapsSm>
+          </VStack>
+          <VStack space={8} style={{flex: 1}}>
+            <AllCapsSmBlack>Expires</AllCapsSmBlack>
+            <AllCapsSm style={{textTransform: 'none'}} color="lightText">
+              {forecast.expires_time}
+            </AllCapsSm>
+          </VStack>
+          <VStack space={8} style={{flex: 1}}>
+            <AllCapsSmBlack>Author</AllCapsSmBlack>
+            <AllCapsSm style={{textTransform: 'none'}} color="lightText">
+              {forecast.author || 'Unknown'}
+              {'\n'}
+            </AllCapsSm>
+          </VStack>
+        </HStack>
+        <VStack alignItems="stretch" pt={4}>
+          {forecast.zones[zone.name].slice(0, 2).map((forecast, index) => {
```

Comment:
Are we guaranteed that `zone.name` is a key in `forecast.zones`?

Reply:
🤷 as guaranteed as we can be given that this is scraping a web page! but I'll add a null check and display the error

Reply:
actually I'll pull this in as part of the splitting into two hooks and adding params